Sources/RInAppMessaging/ViewListener.swift

```diff
@@ -0,0 +1,225 @@
+            return
+        }
+
+        iterateOverDisplayedViews { existingView, identifier, _ in
+            self.observers.forEach {
+                $0.value?.viewDidMoveToWindow(existingView, identifier: identifier)
+            }
+        }
+    }
+
+    func stopListening() {
+        guard isListening else {
+            return
+        }
+
+        isListening = false
+        guard performSwizzling() else {
+            isListening = true
+            assertionFailure()
+            return
+        }
+    }
+
+    func addObserver(_ observer: ViewListenerObserver) {
+        observers.append(WeakWrapper(value: observer))
+    }
+
+    func iterateOverDisplayedViews(_ handler: @escaping (UIView, String, inout Bool) -> Void) {
+        DispatchQueue.main.async {
+            guard let allWindowSubviews = UIApplication.shared.getKeyWindow()?.getAllSubviews() else {
+                return
+            }
+            var stop = false
+            for existingView in allWindowSubviews {
+                guard !stop else {
+                    return
+                }
+                guard let identifier = existingView.accessibilityIdentifier, !identifier.isEmpty else {
+                    continue
+                }
+                handler(existingView, identifier, &stop)
+            }
+        }
+    }
+
+    private func performSwizzling() -> Bool {
+        [swizzle(#selector(UIView.didMoveToSuperview), with: #selector(UIView.swizzled_didMoveToSuperview)),
+         swizzle(#selector(UIView.removeFromSuperview), with: #selector(UIView.swizzled_removeFromSuperview)),
+         swizzle(#selector(setter: UIView.accessibilityIdentifier),
+                 with: #selector(NSObject.swizzled_setAccessibilityIdentifier),
+                 of: NSObject.self),
+//         swizzle(#selector(setter: UILabel.text),
+//                 with: #selector(UILabel.swizzled_setText),
+//                 of: UILabel.self),
+//         swizzle(#selector(setter: UILabel.attributedText),
+//                 with: #selector(UILabel.swizzled_setAttributedText),
+//                 of: UILabel.self),
+//         swizzle(#selector(setter: UITextView.attributedText),
+//                 with: #selector(UITextView.swizzled_setAttributedText),
+//                 of: UITextView.self),
+         swizzle(#selector(UIView.didMoveToWindow), with: #selector(UIView.swizzled_didMoveToWindow))].allSatisfy { $0 == true }
+    }
+
+    private func swizzle(_ sel1: Selector, with sel2: Selector, of class: AnyClass = UIView.self) -> Bool {
+        guard let originalMethod = class_getInstanceMethod(`class`, sel1),
+              let swizzledMethod = class_getInstanceMethod(`class`, sel2) else {
+                  assertionFailure()
+                  return false
+              }
+
+        method_exchangeImplementations(originalMethod, swizzledMethod)
+
+        return true
+    }
+}
+
+private extension NSObject {
```

Comment:
Is this swizzling used for tests purpose?

Reply:
No. it is used for swizzling accessibilityIdentifier which is defined in `UIAccessibilityIdentification` protocol. NSObject is a common class for classes that implement that protocol.